cfg/atmos.cfg

```diff
@@ -1,22 +1,30 @@
 SYMBOLS {
-    __STACKSIZE__: type = weak, value = $0800; # 2k stack
+    __TAPEHDR__:   type = import;
+    __BASHDR__:    type = import;
+    __PROGFLAG__:  type = weak, value = $00; # $00=BASIC, $80=machine code
+    __AUTORUN__:   type = weak, value = $00; # $C7=run, $00=only load
+    __STACKSIZE__: type = weak, value = $0800; # 2K stack
+    __RAMEND__:    type = weak, value = $9800; # graphics RAM not grabbed
+#   __RAMEND__:    type = weak, value = $B400; # graphics RAM grabbed
 }
 MEMORY {
     ZP:      file = "", define = yes, start = $00E2, size = $001A;
-    TAPEHDR: file = %O, type   = ro,  start = $0000, size = $0011;
-    RAM:     file = %O, define = yes, start = $0500, size = $9300 - __STACKSIZE__;
+    TAPEHDR: file = %O, type   = ro,  start = $0000, size = $001F;
+    BASHEAD: file = %O, define = yes, start = $0501, size = $000D;
+    RAM:     file = %O, define = yes, start = __BASHEAD_LAST__, size = __RAMEND__ - __RAM_START__ - __STACKSIZE__;
 }
 SEGMENTS {
+    ZEROPAGE: load = ZP,      type = zp;
     TAPEHDR:  load = TAPEHDR, type = ro;
+    BASHDR:   load = BASHEAD, type = ro,  define = yes, optional = yes;
     STARTUP:  load = RAM,     type = ro;
     LOWCODE:  load = RAM,     type = ro,                optional = yes;
-    INIT:     load = RAM,     type = ro,  define = yes, optional = yes;
     CODE:     load = RAM,     type = ro;
     RODATA:   load = RAM,     type = ro;
+    INIT:     load = RAM,     type = ro,  define = yes, optional = yes;
     DATA:     load = RAM,     type = rw;
-    ZPSAVE:   load = RAM,     type = bss, define = yes;
+    ZPSAVE:   load = RAM,     type = rw,  define = yes;
```

Comment:
Originally, the zpsave buffer was put somewhere in the middle of the `DATA:` segment.  ld65 _filled_ it with the current `fillval`, so that the rest of `DATA:` would be loaded at its proper place.  If that buffer had been put at the end of `DATA:`, then it wouldn't have been filled.  But, we couldn't control where it was put.  That's why it was moved into its own segment.  Now, we _can_ make sure that it is put after the last segment that has stuff written into the program file.  Therefore, ld65 won't fill it; its space isn't in the program binary.  ("type=bss" stops object files from writing into a segment; it doesn't stop ld65 from _padding_ that segment.)

The new Atmos format does write one byte into `ZPSAVE:`; but, nothing is written after that byte.  Therefore, ld65 doesn't fill the rest of `ZPSAVE:`; the other 25 bytes aren't in the binary.  (I double-checked it by compiling a program, loading it into an emulator, and looking at zpsave's RAM in the emulator's monitor.)